source/common/http/conn_manager_utility.cc

```diff
@@ -28,6 +28,12 @@ void ConnectionManagerUtility::mutateRequestHeaders(Http::HeaderMap& request_hea
                                                     const Router::Config& route_config,
                                                     Runtime::RandomGenerator& random,
                                                     Runtime::Loader& runtime) {
+  // Add user-specified headers to the request before scrubbing the request headers.
```

Comment:
I was chatting with Matt in Gitter. I am going to rework this a bit. I initially thought we could enforce some sort of rules on what headers the user can/cannot add. But the enforcement logic was messy. So, the decision was to have the code be simple. If users add bad configs (e.g., override authority/connection headers, etc.), the behavior will be undefined. We will mention this in the docs.